pkg/trace/obfuscate/sql_tokenizer.go

```diff
@@ -246,198 +247,210 @@ func (tkn *SQLTokenizer) skipBlank() {
 	for unicode.IsSpace(tkn.lastChar) {
 		tkn.next()
 	}
+	tkn.grab()
+}
+
+// myFastToUpper returns a copy of the byte slice s with all Unicode letters mapped to
+// their upper case. It uses the array pointed to by dst as the destination if possible.
+func myFastToUpper(src, dst []byte) []byte {
```

Comment:
I think this method name is a bit rough. Just call it `uppercase` and explain why its fast in a comment.

Reply:
When copying code from somewhere, let's make sure we always mention where it is coming from. Along with that an explanation of why the standard library `bytes.ToUpper` is mandatory IMHO. It will help with both understanding and maintaining it.

Lastly, how much more benefit did this bring?

Reply:
Sorry, I meant to rename this before I pushed it, but missed it. 
I will give it a more appropriate name and document why we want to use this instead of the standard library one.

I can put up a benchmark with/without it. It saves an allocation on every identifier.

Reply:
Here is the benchmark for `bytes.ToUpper` vs `myFastToUpper`. `newToUpper` is this PR. `oldToUpper` is this PR plus the following diff:
```diff
--- a/pkg/trace/obfuscate/sql_tokenizer.go
+++ b/pkg/trace/obfuscate/sql_tokenizer.go
@@ -294,8 +294,9 @@ func (tkn *SQLTokenizer) scanIdentifier() (TokenKind, []byte) {
 
 	t := tkn.grab()
 	// Space allows us to upper-case identifiers 200 bytes long or less without allocating.
-	var space [200]byte
-	upper := myFastToUpper(t, space[:0])
+	//var space [200]byte
+	//upper := myFastToUpper(t, space[:0])
+	upper := bytes.ToUpper(t)
 	if keywordID, found := keywords[string(upper)]; found {
 		return keywordID, t
 	}
```

```
% benchcmp oldToUpper.txt newToUpper.txt 
benchmark                                      old ns/op     new ns/op     delta
BenchmarkObfuscateSQLString/Escaping/512-8     8809          7652          -13.13%
BenchmarkObfuscateSQLString/Grouping/199-8     4208          3780          -10.17%
BenchmarkObfuscateSQLString/query1/3694-8      78783         67797         -13.94%
BenchmarkObfuscateSQLString/query3/980-8       20154         18383         -8.79%
BenchmarkObfuscateSQLString/random-8           1220          1305          +6.97%

benchmark                                      old allocs     new allocs     delta
BenchmarkObfuscateSQLString/Escaping/512-8     32             12             -62.50%
BenchmarkObfuscateSQLString/Grouping/199-8     21             10             -52.38%
BenchmarkObfuscateSQLString/query1/3694-8      405            83             -79.51%
BenchmarkObfuscateSQLString/query3/980-8       96             6              -93.75%
BenchmarkObfuscateSQLString/random-8           13             8              -38.46%

benchmark                                      old bytes     new bytes     delta
BenchmarkObfuscateSQLString/Escaping/512-8     2144          1936          -9.70%
BenchmarkObfuscateSQLString/Grouping/199-8     928           816           -12.07%
BenchmarkObfuscateSQLString/query1/3694-8      20336         16256         -20.06%
BenchmarkObfuscateSQLString/query3/980-8       4224          3008          -28.79%
BenchmarkObfuscateSQLString/random-8           288           248           -13.89%
```

For renaming, I chose `toUpper` to be consistent with its source, `bytes.ToUpper`.